### DISCUSSION TITLE
Is it possible to get a best effort solution to the collision free inverse kinematics solver?

### DISCUSSION THREAD

**Discussion by BolunDai0216** [NVlabs/curobo#141] [Software Q&A] [answered]
Hi,

I was wondering if the inverse kinematics solver can give "best-effort" solutions to infeasible problems that minimizes the least square distance between the end-effector pose and a target pose.

Thanks!

**Comment by balakumar-s** [comment]
Do you want the best effort solution to be collision free?

**Reply by BolunDai0216** [reply]
Yes, I want to also be collision free.

**Reply by balakumar-s** [reply, marked as answer]
You can increase the position reaching success threshold to 0.1 (to reach within 10cm). You can increase it further based on your use case.
https://github.com/NVlabs/curobo/blob/88eac64edca2d2b641c1622e63f3e750157596c4/examples/ik_example.py#L46

**Reply by BolunDai0216** [reply]
> You can increase the position reaching success threshold to 0.1 (to reach within 10cm). You can increase it further based on your use case.
> 
> https://github.com/NVlabs/curobo/blob/88eac64edca2d2b641c1622e63f3e750157596c4/examples/ik_example.py#L46 

Thanks!
